### PR TITLE
Databases

### PR DESCRIPTION
**Pull request by jimflores5**
Please pay particular attention to the clarity of the videos, exercise instructions, and project directions.

### CODE REVIEW

File: src/chapters/databases/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+.. admonition:: Warning
+
+   News stories pop up frequently about hackers stealing information or
+   companies accidentally posting their customers' data online.
+   
+   When you attach a database to your web application, you MUST take steps to
+   protect the information. Visitors to your website should never have direct
+   access to where you store sensitive data.
+
```
Comment (codinglikeagirl42) [inline src/chapters/databases/introduction.rst:61]: Really wanted a bit more info- maybe a link or suggestion for a google search for more info to students so inclined?

File: src/chapters/databases/db-practice.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+Code along with this video as we access the information in the database and
+display it in the console.
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube.com/embed/CPHPjP-CRnA" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
```
Comment (codinglikeagirl42) [inline src/chapters/databases/db-practice.rst:100]: In the video you can see you commented out the code from the section above, but students might miss it.  Thinking a note above this section might work, so as to not have to reshoot the video.

File: src/chapters/databases/more-db-practice.rst [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+By using an :ref:`UPDATE <sql-update-syntax>` query, we can change one or more
+entries in a table. Code along with the following video to practice this
+technique.
+
+.. raw:: html
+
+   <section class="vid_box">
+      <iframe class="vid" src="https://www.youtube.com/embed/piXiY9j64I4" frameborder="1" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+   </section>
+
```
Comment (codinglikeagirl42) [inline src/chapters/databases/more-db-practice.rst:19]: There is an extra column and value in this video, (recent grad  yes) but the students haven't added that and might get stuck.  Maybe a short video about adding columns?

File: src/chapters/databases/db-tips.rst [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+   required! Using ``INT`` causes errors.
+#. ``PRIMARY KEY`` sets up the process to automatically assign an integer to
+   ``key_name``. This happens each time a new row is added to the table. We do
+   NOT need to assign a value to ``key_name`` ourselves.
+#. Python keeps track of the largest primary key in the table. When a new row
+   is added, its primary key is assigned the next higher value.
+#. We can assign our own value to ``key_name`` when we ``INSERT`` a row.
+   However, the action will throw an error if the value we choose matches a
+   primary key already in the table.
+
```
Comment (codinglikeagirl42) [inline src/chapters/databases/db-tips.rst:169]: Not sure if you want to mention that you can add a primary key to an existing table, its just more complicated.

File: src/chapters/databases/db-practice.rst (hunk 2 above)
Comment (jimflores5) [reply src/chapters/databases/db-practice.rst:100]: Got it.  I added text to indicate which lines the students should comment out.

File: src/chapters/databases/introduction.rst (hunk 1 above)
Comment (jimflores5) [reply src/chapters/databases/introduction.rst:61]: I thought the same thing, but I couldn't find a good, accessible webpage for beginners.  If you have a suggestion, I'm all ears!

Comment (codinglikeagirl42) [reply src/chapters/databases/introduction.rst:61]: I really liked this guys video: https://www.youtube.com/watch?v=LsHf3JSDBVc and this article was insightful: https://ckraczkowsky.medium.com/building-a-secure-admin-interface-with-flask-admin-and-flask-security-13ae81faa05

File: src/chapters/databases/more-db-practice.rst (hunk 3 above)
Comment (jimflores5) [reply src/chapters/databases/more-db-practice.rst:19]: Good catch!  I totally missed that.

I added an admonition box above the video that gives the students the code they need to run.  Let me know what you think.

File: src/chapters/databases/db-tips.rst (hunk 4 above)
Comment (jimflores5) [reply src/chapters/databases/db-tips.rst:169]: I did consider this when I was writing the page, but I thought that keeping things simple was a better option.

Starting with a PK column is the practice we want to enforce.  If motivated students want to change an existing table, I'm betting they can research it themselves.  ;-)

BTW, `sqlite3` automatically assigns primary keys, even if there's no column in the table for them.  However, that is definitely a rabbit hole I didn't want the kids to follow me down!

File: src/chapters/databases/introduction.rst (hunk 1 above)
Comment (jimflores5) [reply src/chapters/databases/introduction.rst:61]: As per our conversation, I added the standard "beyond the scope of this course" comment.

File: src/chapters/databases/more-db-practice.rst (hunk 3 above)
Comment (codinglikeagirl42) [reply src/chapters/databases/more-db-practice.rst]: Looks Good!

File: src/chapters/databases/db-tips.rst (hunk 4 above)
Comment (codinglikeagirl42) [reply src/chapters/databases/db-tips.rst:169]: I might have to venture down that rabbit hole... thanks for the tip!

**Pull request merged**